### PR TITLE
Add warning about upcoming changes to `MeasurementProcess` `__eq__` and `__hash__

### PR DESCRIPTION
**Pull request by mudit2812**
**Context:**
Operators and measurement processes will have changed behaviour soon for equality and hashing.

**Description of the Change:**
* Add warning to the `__eq__` and `__hash__` dunder methods of `MeasurementProcess` with information about upcoming changes.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -241,9 +241,10 @@ array([False, False])
   been deprecated. Please use `QuantumScript.bind_new_parameters` instead.
   [(#4346)](https://github.com/PennyLaneAI/pennylane/pull/4346)
 
-* `Operator.__eq__` and `Operator.__hash__` will now raise warnings to reflect upcoming
-  changes to operator equality and hashing.
+* The `__eq__` and `__hash__` dunder methods of `Operator` and `MeasurementProcess` will now raise
+  warnings to reflect upcoming changes to operator equality and hashing.
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:245]:
```suggestion
  warnings to reflect upcoming changes to operator and measurement process
  equality and hashing.
```

File: tests/measurements/test_measurements.py [hunk 2]
```diff
@@ -124,6 +124,62 @@ class NoReturnTypeMeasurement(MeasurementProcess):
     assert mp.return_type is None
 
 
+def test_eq_warning():
```
Comment (timmysilv) [inline tests/measurements/test_measurements.py:127]: don't really need this with the more involved testing below

File: tests/measurements/test_measurements.py [hunk 3]
```diff
@@ -124,6 +124,62 @@ class NoReturnTypeMeasurement(MeasurementProcess):
...
+
+    mp1 = DummyMP(0)
+    mp2 = DummyMP(0)
+
+    with pytest.warns(UserWarning, match="The behaviour of measurement process equality"):
+        assert mp1 == mp1  # pylint: disable=comparison-with-itself
+        assert mp1 != mp2
+
+
+def test_hash_warning():
```
Comment (timmysilv) [inline tests/measurements/test_measurements.py:156]: same comment as for eq

**Review by timmysilv — APPROVED** [review]
looks good!

**Pull request merged**